runtime/service/src/main/java/org/apache/polaris/service/catalog/policy/PolicyCatalog.java

```diff
@@ -153,24 +157,46 @@ public Policy createPolicy(
     return constructPolicy(resultEntity);
   }
 
-  public List<PolicyIdentifier> listPolicies(Namespace namespace, PolicyType policyType) {
+  public List<PolicyIdentifier> listPolicies(Namespace namespace, @Nullable PolicyType policyType) {
     PolarisResolvedPathWrapper resolvedEntities = resolvedEntityView.getResolvedPath(namespace);
     if (resolvedEntities == null) {
       throw new IllegalStateException(
           String.format("Failed to fetch resolved namespace '%s'", namespace));
     }
 
-    List<PolarisEntity> catalogPath = resolvedEntities.getRawFullPath();
-    // TODO: when the "policyType" filter is null we should only call "listEntities" instead
+    List<PolarisEntityCore> catalogPath =
+        PolarisEntity.toCoreList(resolvedEntities.getRawFullPath());
+    if (policyType == null) {
+      // without a policyType filter we can call listEntities to acquire the entity names
+      ListEntitiesResult listEntitiesResult =
+          metaStoreManager.listEntities(
```

Comment:
I believe this deserves a test to ensure the results are the consistent on both call paths.

Reply:
i did not mention this but i had checked before that both call paths are already covered by tests [here](https://github.com/apache/polaris/blob/bc5d0c49f788e3520761ba4aae1c19288a60452f/runtime/service/src/test/java/org/apache/polaris/service/catalog/AbstractPolicyCatalogTest.java#L312-L356) afaict

Reply:
Nice :+1: Thx!